analysisservice/src/main/java/nu/yona/server/analysis/rest/AnalysisEngineController.java

```diff
@@ -7,26 +7,54 @@
  *******************************************************************************/
 package nu.yona.server.analysis.rest;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import nu.yona.server.analysis.service.AnalysisEngineService;
+import nu.yona.server.analysis.service.PotentialConflictDTO;
+import nu.yona.server.goals.service.GoalService;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.hateoas.Resource;
+import org.springframework.hateoas.Resources;
+import org.springframework.http.HttpEntity;
 import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
-import nu.yona.server.analysis.service.AnalysisEngineService;
-import nu.yona.server.analysis.service.PotentialConflictDTO;
-
 @Controller
 @RequestMapping(value = "/analysisEngine/")
 public class AnalysisEngineController {
 	@Autowired
 	private AnalysisEngineService analysisEngineService;
 
+	@Autowired
+	private GoalService goalService;
+	
 	@RequestMapping(method = RequestMethod.POST)
 	@ResponseStatus(value = HttpStatus.OK)
 	public void analyze(@RequestBody PotentialConflictDTO potentialConflictPayload) {
 		analysisEngineService.analyze(potentialConflictPayload);
 	}
+	
+	@RequestMapping(value = "/relevantCategories/", method = RequestMethod.GET)
+	@ResponseBody
+	public HttpEntity<CategoriesResource> getRelevantCategories() {
+		CategoriesDTO categories = new CategoriesDTO(goalService.getAllCategories());
```

Comment:
Instead of this and the method ``getAllCategories`` in ``GoalService``, I would use the below two statements:
```
		Set<String> allCategories = goalService.getAllGoals().stream().flatMap(g -> g.getCategories().stream())
				.collect(Collectors.toSet());
		CategoriesDTO categories = new CategoriesDTO(allCategories);
```